otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/zio/ConcurrentMetricRegistryClient.scala

```diff
@@ -32,33 +35,54 @@ object ConcurrentMetricRegistryClient {
     )
   }
 
-  case class MetricHook[In, Out](
-    update: In => Unit,
-    get: () => Out,
-    modify: In => Unit
-  )
-
-  private lazy val runtimeMirror = ru.runtimeMirror(getClass.getClassLoader)
-
-  private lazy val unsafe = {
-    val runtimeMirror  = ru.runtimeMirror(getClass.getClassLoader)
-    val moduleSymbol   = runtimeMirror.staticModule("zio.Unsafe$")
-    val moduleMirror   = runtimeMirror.reflectModule(moduleSymbol)
-    val moduleInstance = moduleMirror.instance
-    val instanceMirror = runtimeMirror.reflect(moduleInstance)
-    val methodSymbol   = instanceMirror.symbol.info.member(ru.TermName("unsafe")).asTerm.getter.asMethod
-    val methodMirror   = instanceMirror.reflectMethod(methodSymbol)
-    methodMirror.apply().asInstanceOf[zio.Unsafe]
+  def addListener(listener: MetricListener): Unit = {
```

Comment:
So this way we: 
 - add our own custom zio metrics listener (a proxy instance)
 - everytime "updateHistogram" is called we can invoke our custom action (in our case update the otel sync histogram)
 - this can be extended with other metrics that we just no-op for now
 
 This is so nice!